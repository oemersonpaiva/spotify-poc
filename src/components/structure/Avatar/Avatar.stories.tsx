import { Meta, Story } from '@storybook/react/types-6-0'
import Avatar, { AvatarProps } from '.'

export default {
  title: 'structure/Avatar',
  component: Avatar
} as Meta

export const Default: Story<AvatarProps> = (args) => <Avatar {...args} />

Default.args = {
  alt: 'Username',
  src: 'https://i.pravatar.cc/300'
}
