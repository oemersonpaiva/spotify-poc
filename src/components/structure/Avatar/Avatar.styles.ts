import styled, { css } from 'styled-components'

export const Wrapper = styled.a`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.primary};
    border-radius: 50%;
    cursor: pointer;
    display: block;
    overflow: hidden;
    height: 3.5rem;
    width: 3.5rem;
  `}
`

export const AvatarImg = styled.img`
  height: 100%;
  width: 100%;
`
