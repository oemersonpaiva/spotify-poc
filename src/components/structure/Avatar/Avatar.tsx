import * as S from './Avatar.styles'

export type AvatarProps = {
  src: string
  alt: string
}

const Avatar = ({ src, alt }: AvatarProps) => (
  <S.Wrapper>
    <S.AvatarImg src={src} alt={alt} />
  </S.Wrapper>
)

export default Avatar
