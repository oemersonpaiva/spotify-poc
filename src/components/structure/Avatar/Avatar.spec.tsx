import { renderWithProviders } from 'utils/tests/helpers'
import Avatar from './Avatar'

describe('<Avatar />', () => {
  it('should render the heading', () => {
    const { getByRole } = renderWithProviders(
      <Avatar src="https://i.pravatar.cc/300" alt="User" />
    )
    const avatarRender = getByRole('img', { name: /user/i })

    expect(avatarRender).toBeInTheDocument()
    expect(avatarRender).toHaveAttribute('alt', 'User')
    expect(avatarRender).toMatchSnapshot()
  })
})
