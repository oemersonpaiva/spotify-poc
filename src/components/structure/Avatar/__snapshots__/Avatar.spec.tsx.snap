// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render the heading 1`] = `
<img
  alt="User"
  class="sc-gsTCUz gbCiAz"
  src="https://i.pravatar.cc/300"
/>
`;
